packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts

```diff
@@ -350,65 +350,53 @@ defineProperties(Element.prototype, {
         },
-        getElementsByTagName: {
-            value(this: HTMLBodyElement): HTMLCollectionOf<Element> {
-                const elements = arrayFromCollection(
-                    elementGetElementsByTagName.apply(this, ArraySlice.call(arguments) as [string])
-                );
+        writable: true,
+        enumerable: true,
+        configurable: true,
+    },
+    getElementsByTagName: {
+        value(this: HTMLBodyElement): HTMLCollectionOf<Element> {
+            const elements = arrayFromCollection(
+                elementGetElementsByTagName.apply(this, ArraySlice.call(arguments) as [string])
+            );
 
-                // Note: we deviate from native shadow here, but are not fixing
-                // due to backwards compat: https://github.com/salesforce/lwc/pull/3103
-                return createStaticHTMLCollection(
-                    getNonPatchedFilteredArrayOfNodes(this, elements)
-                );
-            },
-            writable: true,
-            enumerable: true,
-            configurable: true,
+            // Note: we deviate from native shadow here, but are not fixing
+            // due to backwards compat: https://github.com/salesforce/lwc/pull/3103
+            return createStaticHTMLCollection(getNonPatchedFilteredArrayOfNodes(this, elements));
         },
-        getElementsByTagNameNS: {
-            value(this: HTMLBodyElement): HTMLCollectionOf<Element> {
-                const elements = arrayFromCollection(
-                    elementGetElementsByTagNameNS.apply(
-                        this,
-                        ArraySlice.call(arguments) as [string, string]
-                    )
-                );
+        writable: true,
+        enumerable: true,
+        configurable: true,
+    },
+    getElementsByTagNameNS: {
+        value(this: HTMLBodyElement): HTMLCollectionOf<Element> {
+            const elements = arrayFromCollection(
+                elementGetElementsByTagNameNS.apply(
+                    this,
+                    ArraySlice.call(arguments) as [string, string]
+                )
+            );
 
-                // Note: we deviate from native shadow here, but are not fixing
-                // due to backwards compat: https://github.com/salesforce/lwc/pull/3103
-                return createStaticHTMLCollection(
-                    getNonPatchedFilteredArrayOfNodes(this, elements)
-                );
-            },
-            writable: true,
-            enumerable: true,
-            configurable: true,
+            // Note: we deviate from native shadow here, but are not fixing
+            // due to backwards compat: https://github.com/salesforce/lwc/pull/3103
+            return createStaticHTMLCollection(getNonPatchedFilteredArrayOfNodes(this, elements));
         },
-    });
-}
+        writable: true,
+        enumerable: true,
+        configurable: true,
+    },
+});
```

Comment:
I traced this back to 2019: https://github.com/salesforce/lwc/pull/1614

If this is still a problem in Jest, I assume our Nucleus tests will tell us soon enough. I'd also like to add an explicit test for Jest here.

Reply:
Also people can disable synthetic shadow in Jest now if they want.

Reply:
Welp Nucleus failed, so maybe there's our answer. Let's try reverting this part.

Reply:
Nucleus passes now, so apparently it is not safe to change this. 😞

Reply:
Yes, it's a long-standing JSDOM issue. The querySelector library used by JSDOM operates on the userland DOM objects (the one patched by user-land code) and not the internal objects. It attempted in the past to fix this, but it was a lot more work than expected...

Reply:
Instead of doing a `process.env` check here, another approach would be to check if the LWC engine is evaluated in the context of JSDOM.

Reply:
Based on [this SO thread](https://stackoverflow.com/questions/50940640/how-to-determine-if-jest-is-running-the-code-or-not/52231746), there does not seem to be a bulletproof way of checking whether we are running in Jest or not. I'd suggest we just stick with `process.env.NODE_ENV === 'test'`.